### PR TITLE
Fixing StatusCode==0 for readMany

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

CosmosDiagnosticsContext.getStatusCode always returns 0 for readMany. This PR fixes that and adds regression coverage.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java [hunk 1]
```diff
@@ -2896,10 +2896,20 @@ public <T> Mono<FeedResponse<T>> readMany(
                                                aggregatedDiagnostics, aggregateRequestStatistics);
 
                                            state.mergeDiagnosticsContext();
+                                           CosmosDiagnosticsContext ctx = state.getDiagnosticsContextSnapshot();
+                                           ctxAccessor.recordOperation(
+                                               ctx,
+                                               200,
+                                               0,
+                                                finalList.size(),
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:2904]: nit:indent

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @FabianMeiswinkel

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java]: Fixed

**Pull request merged**